kafka/src/main/java/io/cloudevents/kafka/KafkaTransportUtil.java

```diff
@@ -0,0 +1,109 @@
+        headers.add(new RecordHeader(transportHeaders.sourceKey(), ((Serde) Serdes.serdeFrom(String.class)).serializer().serialize(null, cloudEvent.getSource().toString())));
+        headers.add(new RecordHeader(transportHeaders.idKey(), ((Serde) Serdes.serdeFrom(String.class)).serializer().serialize(null, cloudEvent.getId())));
+
+        if (cloudEvent.getSchemaURL().isPresent()) {
+            headers.add(new RecordHeader(transportHeaders.schemaUrlKey(), ((Serde) Serdes.serdeFrom(String.class)).serializer().serialize(null, cloudEvent.getSchemaURL().get().toString()) ));
+        }
+
+        if (cloudEvent.getTime().isPresent()) {
+            headers.add(new RecordHeader(transportHeaders.timeKey(), ((Serde) Serdes.serdeFrom(String.class)).serializer().serialize(null, cloudEvent.getTime().get().toString()) ));
+        }
+        //todo: extensions... they are optional...
+
+        return headers;
+    }
+
+    /**
+     * Create a CloudEvent from a message consumed from a Kafka topic. Populates the CloudEvent attributes
+     * from the Kafka Headers and the data from the Kafka value.
+     * @param record message receieved from Kafka
+     * @param <K> Message Key
+     * @param <V> Message Value
+     * @return CloudEvent representation of the Kafka message
+     */
+    public static <K, V> CloudEvent<Map<K, V>> createFromConsumerRecord(final ConsumerRecord<K, V> record) {
+
+        final Headers headers = record.headers();
+        final CloudEventBuilder builder = new CloudEventBuilder();
+
+        try {
+            builder.type(Serdes.serdeFrom(String.class).deserializer().deserialize(null, headers.lastHeader(transportHeaders.typeKey()).value()));
+            builder.source(URI.create(Serdes.serdeFrom(String.class).deserializer().deserialize(null, headers.lastHeader(transportHeaders.sourceKey()).value())));
+            builder.id(Serdes.serdeFrom(String.class).deserializer().deserialize(null, headers.lastHeader(transportHeaders.idKey()).value()));
+
+            if (headers.lastHeader(transportHeaders.timeKey()) != null) {
+                builder.time(ZonedDateTime.parse(Serdes.serdeFrom(String.class).deserializer().deserialize(null, headers.lastHeader(transportHeaders.timeKey()).value()), ISO_ZONED_DATE_TIME));
+            }
+
+            if (headers.lastHeader(transportHeaders.schemaUrlKey()) != null) {
+                builder.schemaURL(URI.create(Serdes.serdeFrom(String.class).deserializer().deserialize(null, headers.lastHeader(transportHeaders.schemaUrlKey()).value())));
+            }
+
+            //todo: add extensions
+
+            // use the pure key/value as the 'data' part
+            final Map<K, V> rawKafkaRecord = new HashMap();
+            rawKafkaRecord.put(record.key(), record.value());
+            builder.data(rawKafkaRecord);
+
+            // we hard wire to this type:
+            builder.contentType("application/kafka"); // todo: move to constant
```

Comment:
What does it mean to have a content type of "application/kafka"? Isn't the transport an orthogonal concern to the content type?